src/core/libmaven/PeakDetector.h

```diff
@@ -138,17 +138,7 @@ class PeakDetector {
 
     static vector<EIC*> pullEICs(mzSlice* slice,
                                  std::vector<mzSample*>& samples,
-                                 int peakDetect,
-                                 int smoothingWindow,
-                                 int smoothingAlgorithm,
-                                 float amuQ1,
-                                 float amuQ3,
-                                 EIC::BaselineMode baselineMode,
-                                 int firstBaselineParameter,
-                                 int secondBaselineParameter,
-                                 double minSignalBaselineDifference,
-                                 int eicType,
-                                 string filterline);
+                                 MavenParameters* mp);
```

Comment:
I have concerns around sending a reference to MavenParameters. The object is shared with multiple classes which can update the parameters. A lot of our UI currently blocks any change in baseline parameters while the algo is in process, so we can work on this in the next release but just pointing it out for future ref.

Reply:
If we  want MavenParameters to be Immutable inside, then we should use a `const reference`.